debian/manpage.1
.\"                                      Hey, EMACS: -*- nroff -*-
.\" (C) Copyright 2014 Shell.Xu <shell909090@gmail.com>,
.\"
.TH SHADOWSOCKS SECTION "August 23, 2014"
.SH NAME
shadowsocks \- Fast tunnel proxy that helps you bypass firewalls
.SH SYNOPSIS
.B ssserver
.RI [ options ]
.br
.B sslocal
.RI [ options ]
.SH DESCRIPTION
shadowsocks is a tunnel proxy helps you bypass firewall. 
.B ssserver
is the server part, and
.B sslocal
is the local part.
.SH OPTIONS
.TP
.B \-h, \-\-help
Show this help message and exit.
.TP
.B \-s SERVER_ADDR
Server address, default: 0.0.0.0.
.TP
.B \-p SERVER_PORT
Server port, default: 8388.
.TP
.B \-k PASSWORD
Password.
.TP
.B \-m METHOD
Encryption method, default: aes-256-cfb.
.TP
.B \-t TIMEOUT
Timeout in seconds, default: 300.
.TP
.B \-c CONFIG
Path to config file.
.TP
.B \-\-fast-open
Use TCP_FASTOPEN, requires Linux 3.7+.
.TP
.B \-\-workers WORKERS
Number of workers, available on Unix/Linux.
.TP
.B \-v, \-vv
Verbose mode.
.TP
.B \-q, \-qq
Quiet mode, only show warnings/errors.
.SH SEE ALSO
.br
The programs are documented fully by
.IR "Shell Xu <shell909090@gmail.com>"
and 
.IR "Clowwindy <clowwindy42@gmail.com>",
available via the Info system.


CHANGES
2.1.0 2014-08-10
- Use only IPv4 DNS server
- Does not ship config.json
- Better error message

2.0.12 2014-07-26
- Support -q quiet mode
- Exit 0 when showing help with -h

2.0.11 2014-07-12
- Prefers IP addresses over hostnames, more friendly with socksify and openvpn

2.0.10 2014-07-11
- Fix UDP on local

2.0.9 2014-07-06
- Fix EWOULDBLOCK on Windows
- Fix Unicode config problem on some platforms

2.0.8 2014-06-23
- Use multiple DNS to query hostnames

2.0.7 2014-06-21
- Fix fastopen on local
- Fallback when fastopen is not available
- Add verbose logging mode -vv
- Verify if hostname is valid

2.0.6 2014-06-19
- Fix CPU 100% on POLL_HUP
- More friendly logging

2.0.5 2014-06-18
- Support a simple config format for multiple ports

2.0.4 2014-06-12
- Fix worker master

2.0.3 2014-06-11
- Fix table encryption with UDP

2.0.2 2014-06-11
- Add asynchronous DNS in TCP relay

2.0.1 2014-06-05
- Better logging
- Maybe fix bad file descriptor

2.0 2014-06-05
- Use a new event model
- Remove gevent
- Refuse to use default password
- Fix a problem when using multiple passwords with table encryption

1.4.5 2014-05-24
- Add timeout in TCP server
- Close sockets in master process

1.4.4 2014-05-17
- Support multiple workers

1.4.3 2014-05-13
- Fix Windows

1.4.2 2014-05-10
- Add salsa20-ctr cipher

1.4.1 2014-05-03
- Fix error log
- Fix EINPROGESS with some version of gevent

1.4.0 2014-05-02
- Adds UDP relay
- TCP fast open support on Linux 3.7+

1.3.7 2014-04-10
- Fix a typo in help

1.3.6 2014-04-10
- Fix a typo in help

1.3.5 2014-04-07
- Add help
- Change default local binding address into 127.0.0.1

1.3.4 2014-02-17
- Fix a bug when no config file exists
- Client now support multiple server ports and multiple server/port pairs
- Better error message with bad config.json format and wrong password

1.3.3 2013-07-09
- Fix default key length of rc2

1.3.2 2013-07-04
- Server will listen at server IP specified in config
- Check config file and show some warning messages

1.3.1 2013-06-29
- Fix -c arg

1.3.0 2013-06-22
- Move to pypi

1.2.3 2013-06-14
- add bind address

1.2.2 2013-05-31
- local can listen at ::0 with -6 arg; bump 1.2.2

1.2.1 2013-05-23
- Fix an OpenSSL crash

1.2 2013-05-22
- Use random iv, we finally have strong encryption

1.1.1 2013-05-21
- Add encryption, AES, blowfish, etc.

1.1 2013-05-16
- Support IPv6 addresses (type 4)
- Drop Python 2.5 support

1.0 2013-04-03
- Fix -6 IPv6

0.9.4 2013-03-04
- Support Python 2.5

0.9.3 2013-01-14
- Fix conn termination null data

0.9.2 2013-01-05
- Change default timeout

0.9.1 2013-01-05
- Add Travis-CI test

0.9 2012-12-30
- Replace send with sendall, fix FreeBSD

0.6 2012-12-06
- Support args

0.5 2012-11-08
- Fix encryption with negative md5sum

0.4 2012-11-02
- Move config into a JSON file
- Auto-detect config path

0.3 2012-06-06
- Move socks5 negotiation to local

0.2 2012-05-11
- Add -6 arg for IPv6
- Fix socket.error

0.1 2012-04-20
- Initial version


debian/manpage.xml.ex
<?xml version='1.0' encoding='UTF-8'?>
<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook XML V4.5//EN"
"http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd" [

<!--

`xsltproc -''-nonet \
          -''-param man.charmap.use.subset "0" \
          -''-param make.year.ranges "1" \
          -''-param make.single.year.ranges "1" \
          /usr/share/xml/docbook/stylesheet/docbook-xsl/manpages/docbook.xsl \
          manpage.xml'

A manual page <package>.<section> will be generated. You may view the
manual page with: nroff -man <package>.<section> | less'. A typical entry
in a Makefile or Makefile.am is:

DB2MAN = /usr/share/sgml/docbook/stylesheet/xsl/docbook-xsl/manpages/docbook.xsl
XP     = xsltproc -''-nonet -''-param man.charmap.use.subset "0"

manpage.1: manpage.xml
        $(XP) $(DB2MAN) $<

The xsltproc binary is found in the xsltproc package. The XSL files are in
docbook-xsl. A description of the parameters you can use can be found in the
docbook-xsl-doc-* packages. Please remember that if you create the nroff
version in one of the debian/rules file targets (such as build), you will need
to include xsltproc and docbook-xsl in your Build-Depends control field.
Alternatively use the xmlto command/package. That will also automatically
pull in xsltproc and docbook-xsl.

Notes for using docbook2x: docbook2x-man does not automatically create the
AUTHOR(S) and COPYRIGHT sections. In this case, please add them manually as
<refsect1> ... </refsect1>.

To disable the automatic creation of the AUTHOR(S) and COPYRIGHT sections
read /usr/share/doc/docbook-xsl/doc/manpages/authors.html. This file can be
found in the docbook-xsl-doc-html package.

Validation can be done using: `xmllint -''-noout -''-valid manpage.xml`

General documentation about man-pages and man-page-formatting:
man(1), man(7), http://www.tldp.org/HOWTO/Man-Page/

-->

  <!-- Fill in your name for FIRSTNAME and SURNAME. -->
  <!ENTITY dhfirstname "FIRSTNAME">
  <!ENTITY dhsurname   "SURNAME">
  <!-- dhusername could also be set to "&dhfirstname; &dhsurname;". -->
  <!ENTITY dhusername  "Shell.Xu">
  <!ENTITY dhemail     "shell909090@gmail.com">
  <!-- SECTION should be 1-8, maybe w/ subsection other parameters are
       allowed: see man(7), man(1) and
       http://www.tldp.org/HOWTO/Man-Page/q2.html. -->
  <!ENTITY dhsection   "SECTION">
  <!-- TITLE should be something like "User commands" or similar (see
       http://www.tldp.org/HOWTO/Man-Page/q2.html). -->
  <!ENTITY dhtitle     "shadowsocks User Manual">
  <!ENTITY dhucpackage "SHADOWSOCKS">
  <!ENTITY dhpackage   "shadowsocks">
]>

<refentry>
  <refentryinfo>
    <title>&dhtitle;</title>
    <productname>&dhpackage;</productname>
    <authorgroup>
      <author>
       <firstname>&dhfirstname;</firstname>
        <surname>&dhsurname;</surname>
        <contrib>Wrote this manpage for the Debian system.</contrib>
        <address>
          <email>&dhemail;</email>
        </address>
      </author>
    </authorgroup>
    <copyright>
      <year>2007</year>
      <holder>&dhusername;</holder>
    </copyright>
    <legalnotice>
      <para>This manual page was written for the Debian system
        (and may be used by others).</para>
      <para>Permission is granted to copy, distribute and/or modify this
        document under the terms of the GNU General Public License,
        Version 2 or (at your option) any later version published by
        the Free Software Foundation.</para>
      <para>On Debian systems, the complete text of the GNU General Public
        License can be found in
        <filename>/usr/share/common-licenses/GPL</filename>.</para>
    </legalnotice>
  </refentryinfo>
  <refmeta>
    <refentrytitle>&dhucpackage;</refentrytitle>
    <manvolnum>&dhsection;</manvolnum>
  </refmeta>
  <refnamediv>
    <refname>&dhpackage;</refname>
    <refpurpose>program to do something</refpurpose>
  </refnamediv>
  <refsynopsisdiv>
    <cmdsynopsis>
      <command>&dhpackage;</command>
      <!-- These are several examples, how syntaxes could look -->
      <arg choice="plain"><option>-e <replaceable>this</replaceable></option></arg>
      <arg choice="opt"><option>--example=<parameter>that</parameter></option></arg>
      <arg choice="opt">
        <group choice="req">
          <arg choice="plain"><option>-e</option></arg>
          <arg choice="plain"><option>--example</option></arg>
        </group>
        <replaceable class="option">this</replaceable>
      </arg>
      <arg choice="opt">
        <group choice="req">
          <arg choice="plain"><option>-e</option></arg>
          <arg choice="plain"><option>--example</option></arg>
        </group>
        <group choice="req">
          <arg choice="plain"><replaceable>this</replaceable></arg>
          <arg choice="plain"><replaceable>that</replaceable></arg>
        </group>
      </arg>
    </cmdsynopsis>
    <cmdsynopsis>
      <command>&dhpackage;</command>
      <!-- Normally the help and version options make the programs stop
           right after outputting the requested information. -->
      <group choice="opt">
        <arg choice="plain">
          <group choice="req">
            <arg choice="plain"><option>-h</option></arg>
            <arg choice="plain"><option>--help</option></arg>
          </group>
        </arg>
        <arg choice="plain">
          <group choice="req">
            <arg choice="plain"><option>-v</option></arg>
            <arg choice="plain"><option>--version</option></arg>
          </group>
        </arg>
      </group>
    </cmdsynopsis>
  </refsynopsisdiv>
  <refsect1 id="description">
    <title>DESCRIPTION</title>
    <para>This manual page documents briefly the
      <command>&dhpackage;</command> and <command>bar</command>
      commands.</para>
    <para>This manual page was written for the Debian distribution
      because the original program does not have a manual page.
      Instead, it has documentation in the GNU <citerefentry>
        <refentrytitle>info</refentrytitle>
        <manvolnum>1</manvolnum>
      </citerefentry> format; see below.</para>
    <para><command>&dhpackage;</command> is a program that...</para>
  </refsect1>
  <refsect1 id="options">
    <title>OPTIONS</title>
    <para>The program follows the usual GNU command line syntax,
      with long options starting with two dashes (`-').  A summary of
      options is included below.  For a complete description, see the
      <citerefentry>
        <refentrytitle>info</refentrytitle>
        <manvolnum>1</manvolnum>
      </citerefentry> files.</para>
    <variablelist>
      <!-- Use the variablelist.term.separator and the
           variablelist.term.break.after parameters to
           control the term elements. -->
      <varlistentry>
        <term><option>-e <replaceable>this</replaceable></option></term>
        <term><option>--example=<replaceable>that</replaceable></option></term>
        <listitem>
          <para>Does this and that.</para>
        </listitem>
      </varlistentry>
      <varlistentry>
        <term><option>-h</option></term>
        <term><option>--help</option></term>
        <listitem>
          <para>Show summary of options.</para>
        </listitem>
      </varlistentry>
      <varlistentry>
        <term><option>-v</option></term>
        <term><option>--version</option></term>
        <listitem>
          <para>Show version of program.</para>
        </listitem>
      </varlistentry>
    </variablelist>
  </refsect1>
  <refsect1 id="files">
    <title>FILES</title>
    <variablelist>
      <varlistentry>
        <term><filename>/etc/foo.conf</filename></term>
        <listitem>
          <para>The system-wide configuration file to control the
            behaviour of <application>&dhpackage;</application>. See
            <citerefentry>
              <refentrytitle>foo.conf</refentrytitle>
              <manvolnum>5</manvolnum>
            </citerefentry> for further details.</para>
        </listitem>
      </varlistentry>
      <varlistentry>
        <term><filename>${HOME}/.foo.conf</filename></term>
        <listitem>
          <para>The per-user configuration file to control the
             behaviour of <application>&dhpackage;</application>. See
             <citerefentry>
               <refentrytitle>foo.conf</refentrytitle>
               <manvolnum>5</manvolnum>
             </citerefentry> for further details.</para>
        </listitem>
      </varlistentry>
    </variablelist>
  </refsect1>
  <refsect1 id="environment">
    <title>ENVIRONMENT</title>
    <variablelist>
      <varlistentry>
        <term><envar>FOO_CONF</envar></term>
        <listitem>
          <para>If used, the defined file is used as configuration
            file (see also <xref linkend="files"/>).</para>
        </listitem>
      </varlistentry>
    </variablelist>
  </refsect1>
  <refsect1 id="diagnostics">
    <title>DIAGNOSTICS</title>
    <para>The following diagnostics may be issued
      on <filename class="devicefile">stderr</filename>:</para>
    <variablelist>
      <varlistentry>
        <term><errortext>Bad configuration file. Exiting.</errortext></term>
        <listitem>
          <para>The configuration file seems to contain a broken configuration
            line. Use the <option>--verbose</option> option, to get more info.
          </para>
        </listitem>
      </varlistentry>
    </variablelist>
    <para><command>&dhpackage;</command> provides some return codes, that can
      be used in scripts:</para>
    <segmentedlist>
      <segtitle>Code</segtitle>
      <segtitle>Diagnostic</segtitle>
      <seglistitem>
        <seg><errorcode>0</errorcode></seg>
        <seg>Program exited successfully.</seg>
      </seglistitem>
      <seglistitem>
        <seg><errorcode>1</errorcode></seg>
        <seg>The configuration file seems to be broken.</seg>
      </seglistitem>
    </segmentedlist>
  </refsect1>
  <refsect1 id="bugs">
    <!-- Or use this section to tell about upstream BTS. -->
    <title>BUGS</title>
    <para>The program is currently limited to only work
      with the <package>foobar</package> library.</para>
    <para>The upstreams <acronym>BTS</acronym> can be found
      at <ulink url="http://bugzilla.foo.tld"/>.</para>
  </refsect1>
  <refsect1 id="see_also">
    <title>SEE ALSO</title>
    <!-- In alpabetical order. -->
    <para><citerefentry>
        <refentrytitle>bar</refentrytitle>
        <manvolnum>1</manvolnum>
      </citerefentry>, <citerefentry>
        <refentrytitle>baz</refentrytitle>
        <manvolnum>1</manvolnum>
      </citerefentry>, <citerefentry>
        <refentrytitle>foo.conf</refentrytitle>
        <manvolnum>5</manvolnum>
      </citerefentry></para>
    <para>The programs are documented fully by <citetitle>The Rise and
      Fall of a Fooish Bar</citetitle> available via the <citerefentry>
        <refentrytitle>info</refentrytitle>
        <manvolnum>1</manvolnum>
      </citerefentry> system.</para>
  </refsect1>
</refentry>

